source/fem-dem/gls_vans.cc

```diff
@@ -533,97 +582,202 @@ GLSVANSSolver<dim>::quadrature_centered_sphere_method(bool load_balance_step)
           for (auto &particle : pic)
             {
               auto         particle_properties = particle.get_properties();
               const double r_particle =
-                particle_properties[DEM::PropertiesIndex::dp] * 0.5;
+                0.5 * particle_properties[DEM::PropertiesIndex::dp];
 
               // Loop over neighboring cells to determine if a given neighboring
               // particle contributes to the solid volume of the current
               // reference sphere
               //***********************************************************************
               for (unsigned int n = 0; n < active_neighbors.size(); n++)
                 {
-                  // Define the volume of the reference sphere to be used as the
-                  // averaging volume for the QCM
-                  double reference_sphere_volume;
-                  if (this->cfd_dem_simulation_parameters.void_fraction
-                        ->qcm_sphere_diameter < 1e-16)
+                  // Define the radius of the reference sphere to be used as the
+                  // averaging volume for the QCM, if the reference sphere
+                  // diameter was given by the user the value is already defined
+                  // since is it not dependent on any measure of the active cell
+                  if (calculate_reference_sphere_radius)
                     {
-                      // Volume of sphere equals volume of cell
                       if (this->cfd_dem_simulation_parameters.void_fraction
                             ->qcm_sphere_equal_cell_volume == true)
-                        reference_sphere_volume =
-                          active_neighbors[n]->measure();
+                        {
+                          // Get the radius by the volume of sphere which is
+                          // equal to the volume of cell
+                          R_sphere =
+                            0.5 *
+                            pow(2 * dim * active_neighbors[n]->measure() / M_PI,
+                                1. / dim);
+                        }
                       else
-                        // Volume of sphere based on R_s = h_omega
-                        reference_sphere_volume =
-                          M_PI *
-                          pow((2 *
-                               pow(active_neighbors[n]->measure(), 1. / dim)),
-                              dim) /
-                          (2 * dim);
+                        {
+                          // The radius is obtained from the volume of sphere
+                          // based on R_s = h_omega
+                          double reference_sphere_volume =
+                            M_PI *
+                            pow((2 *
+                                 pow(active_neighbors[n]->measure(), 1. / dim)),
+                                dim) /
+                            (2 * dim);
```

Comment:
Yes

Reply:
Actually, can I do it for 1./dim? Because it won't compile with 1./dim, but with 1/dim. But it won't give the good value (int divided by a int). Am I right?

Reply:
I changed to fixed_power<dim> everything that was ^dim